pkg/cloudprovider/gcp/object_store.go

```diff
@@ -76,16 +91,23 @@ func (o *objectStore) Init(config map[string]string) error {
 	}
 	o.client = client
 
+	o.retriever = &objectRetriever{client: o.client}
+
 	return nil
 }
 
 func (o *objectStore) PutObject(bucket string, key string, body io.Reader) error {
-	w := o.client.Bucket(bucket).Object(key).NewWriter(context.Background())
-	defer w.Close()
+	w := o.retriever.getWriteCloser(bucket, key)
 
+	// The writer returned by NewWriter is asynchronous, so errors aren't guaranteed
+	// until Close() is called
 	_, err := io.Copy(w, body)
+	if err != nil {
+		return err
```

Comment:
I don't think we want to return early here. We still should try to close the writer.

Reply:
Yeah, good catch. I'll update it.